[CI] ShardsCapacityHealthIndicatorServiceIT testGreen failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/hbwvejpcpjcfw/tests/:server:internalClusterTest/org.elasticsearch.health.node.ShardsCapacityHealthIndicatorServiceIT/testGreen

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.health.node.ShardsCapacityHealthIndicatorServiceIT.testGreen" -Dtests.seed=6E24160DEED7B20F -Dtests.locale=de-AT -Dtests.timezone=Pacific/Noumea -Druntime.java=18
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.health.node.ShardsCapacityHealthIndicatorServiceIT&tests.test=testGreen

**Failure excerpt:**
```
java.lang.AssertionError: expected:<UNKNOWN> but was:<GREEN>

  at __randomizedtesting.SeedInfo.seed([6E24160DEED7B20F:8DEC396855CA327D]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:118)
  at org.junit.Assert.assertEquals(Assert.java:144)
  at org.elasticsearch.health.node.ShardsCapacityHealthIndicatorServiceIT.testGreen(ShardsCapacityHealthIndicatorServiceIT.java:58)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
@@ -90,9 +90,9 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         var healthNode = ESIntegTestCase.waitAndGetHealthNode(internalCluster);
         assertNotNull(healthNode);
 
-        waitForShardLimitsMetadata();
-
         var randomNode = internalCluster.getRandomNodeName();
+        waitForShardLimitsMetadata(randomNode);
+
         var healthService = internalCluster.getInstance(HealthService.class, randomNode);
         var healthIndicatorResults = getHealthServiceResults(healthService, randomNode);
         assertThat(healthIndicatorResults, hasSize(1));
@@ -117,9 +117,9 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         return resultListReference.get();
     }
 
-    private void waitForShardLimitsMetadata() throws Exception {
+    private void waitForShardLimitsMetadata(String node) throws Exception {
         assertBusy(() -> {
-            var healthMetadata = HealthMetadata.getFromClusterState(internalCluster().clusterService().state());
+            var healthMetadata = HealthMetadata.getFromClusterState(internalCluster().clusterService(node).state());
 
             assertNotNull(healthMetadata);
             assertNotNull(healthMetadata.getShardLimitsMetadata());

--- a/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
@@ -16,14 +16,12 @@ import org.elasticsearch.health.HealthStatus;
 import org.elasticsearch.health.metadata.HealthMetadata;
 import org.elasticsearch.test.ESIntegTestCase;
 import org.elasticsearch.test.InternalTestCluster;
-import org.elasticsearch.test.NodeRoles;
 import org.junit.After;
 import org.junit.Before;
 
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 
-import static org.elasticsearch.cluster.node.DiscoveryNodeRole.DATA_FROZEN_NODE_ROLE;
 import static org.elasticsearch.indices.ShardLimitValidator.SETTING_CLUSTER_MAX_SHARDS_PER_NODE;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.hasSize;
@@ -54,6 +52,9 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         // index: 4 shards + 1 replica = 8 shards used (30 - 8 = 22 > 10 available shards)
         createIndex(4, 1);
 
+        ensureStableCluster(internalCluster.getNodeNames().length);
+        waitForHealthMetadata();
+
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.GREEN);
         assertEquals(result.symptom(), "The cluster has enough room to add new shards.");
@@ -65,6 +66,9 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         // index: 11 shards + 1 replica = 22 shards used (30 - 22 < 10 available shards)
         createIndex(10, 1);
 
+        ensureStableCluster(internalCluster.getNodeNames().length);
+        waitForHealthMetadata();
+
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.YELLOW);
         assertEquals(result.symptom(), "Cluster is close to reaching the configured maximum number of shards for data nodes.");
@@ -75,6 +79,9 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
     public void testRed() throws Exception {
         // index: 13 shards + 1 replica = 26 shards used (30 - 26 < 5 available shards)
         createIndex(13, 1);
+
+        ensureStableCluster(internalCluster.getNodeNames().length);
+        waitForHealthMetadata();
 
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.RED);
@@ -95,14 +102,6 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         var healthIndicatorResults = getHealthServiceResults(healthService, healthNodeName);
         assertThat(healthIndicatorResults, hasSize(1));
         return healthIndicatorResults.get(0);
-    }
-
-    private void setUpCluster(InternalTestCluster internalCluster) throws Exception {
-        internalCluster.startMasterOnlyNode();
-        internalCluster.startDataOnlyNode();
-        internalCluster.startNode(NodeRoles.onlyRole(DATA_FROZEN_NODE_ROLE));
-        ensureStableCluster(internalCluster.getNodeNames().length);
-        waitForHealthMetadata();
     }
 
     private List<HealthIndicatorResult> getHealthServiceResults(HealthService healthService, String node) throws Exception {

--- a/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java
@@ -52,9 +52,6 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         // index: 4 shards + 1 replica = 8 shards used (30 - 8 = 22 > 10 available shards)
         createIndex(4, 1);
 
-        ensureStableCluster(internalCluster.getNodeNames().length);
-        waitForHealthMetadata();
-
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.GREEN);
         assertEquals(result.symptom(), "The cluster has enough room to add new shards.");
@@ -66,9 +63,6 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         // index: 11 shards + 1 replica = 22 shards used (30 - 22 < 10 available shards)
         createIndex(10, 1);
 
-        ensureStableCluster(internalCluster.getNodeNames().length);
-        waitForHealthMetadata();
-
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.YELLOW);
         assertEquals(result.symptom(), "Cluster is close to reaching the configured maximum number of shards for data nodes.");
@@ -79,9 +73,6 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
     public void testRed() throws Exception {
         // index: 13 shards + 1 replica = 26 shards used (30 - 26 < 5 available shards)
         createIndex(13, 1);
-
-        ensureStableCluster(internalCluster.getNodeNames().length);
-        waitForHealthMetadata();
 
         var result = fetchShardsCapacityIndicatorResult(internalCluster);
         assertEquals(result.status(), HealthStatus.RED);
@@ -95,11 +86,15 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
     }
 
     private HealthIndicatorResult fetchShardsCapacityIndicatorResult(InternalTestCluster internalCluster) throws Exception {
+        ensureStableCluster(internalCluster.getNodeNames().length);
         var healthNode = ESIntegTestCase.waitAndGetHealthNode(internalCluster);
         assertNotNull(healthNode);
-        var healthNodeName = healthNode.getName();
-        var healthService = internalCluster.getInstance(HealthService.class, healthNodeName);
-        var healthIndicatorResults = getHealthServiceResults(healthService, healthNodeName);
+
+        waitForShardLimitsMetadata();
+
+        var randomNode = internalCluster.getRandomNodeName();
+        var healthService = internalCluster.getInstance(HealthService.class, randomNode);
+        var healthIndicatorResults = getHealthServiceResults(healthService, randomNode);
         assertThat(healthIndicatorResults, hasSize(1));
         return healthIndicatorResults.get(0);
     }
@@ -122,7 +117,7 @@ public class ShardsCapacityHealthIndicatorServiceIT extends ESIntegTestCase {
         return resultListReference.get();
     }
 
-    private void waitForHealthMetadata() throws Exception {
+    private void waitForShardLimitsMetadata() throws Exception {
         assertBusy(() -> {
             var healthMetadata = HealthMetadata.getFromClusterState(internalCluster().clusterService().state());
 

